plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -825,38 +815,57 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
+                                                                [this, is_transient, &next, &trx](fc::exception_ptr ex) {
+                                                                   this->log_trx_results(trx, nullptr, ex, 0, is_transient);
+                                                                   next(std::move(ex));
+                                                                };
+                                                        try {
+                                                           std::rethrow_exception(ex_ptr);
+                                                        } CATCH_AND_CALL(ex_handler)
+                                                     });
+                               return;
+                            }
+
+                            app().executor().post(priority::low, exec_queue::read_write,
+                                                  [this, trx_meta{std::move(trx_meta)}, api_trx, is_transient, next{std::move(next)}, return_failure_traces]() mutable {
+                                                     auto start       = fc::time_point::now();
+                                                     auto idle_time   = this->_time_tracker.add_idle_time(start);
+                                                     auto trx_tracker = this->_time_tracker.start_trx(is_transient, start);
+                                                     fc_tlog(_log, "Time since last trx: ${t}us", ("t", idle_time));
+
+                                                     auto exception_handler =
+                                                             [this, is_transient, &next, &trx_meta](fc::exception_ptr ex) {
+                                                                this->log_trx_results(trx_meta->packed_trx(), nullptr, ex, 0, is_transient);
+                                                                next(std::move(ex));
+                                                             };
+                                                     try {
+                                                        if (!this->process_incoming_transaction_async(trx_meta, api_trx, return_failure_traces, trx_tracker, next)) {
+                                                           if (this->in_producing_mode()) {
+                                                              this->schedule_maybe_produce_block(true);
+                                                           } else {
+                                                              this->restart_speculative_block();
+                                                           }
+                                                        }
+                                                     }
+                                                     CATCH_AND_CALL(exception_handler);
+                                                  });
+                         } );
```

Comment:
I think it would be better to remove all occurences of this-> in this code.

Reply:
Removed